code/modules/research/destructive_analyzer.dm

```diff
@@ -1,81 +1,138 @@
-/*
-Destructive Analyzer
-It is used to destroy hand-held objects and advance technological research. Controls are in the linked R&D console.
-Note: Must be placed within 3 tiles of the R&D Console
-*/
-
 /obj/machinery/r_n_d/destructive_analyzer
-	name = "destructive analyzer"
+	name = "Destructive Analyzer"
 	icon_state = "d_analyzer"
 	var/obj/item/weapon/loaded_item = null
 	var/decon_mod = 0
 
-	use_power = 1
-	idle_power_usage = 30
-	active_power_usage = 2500
-
-/obj/machinery/r_n_d/destructive_analyzer/New(var/atom/location, var/direction, var/nocircuit = FALSE)
-	..()
+/obj/machinery/r_n_d/destructive_analyzer/Initialize()
+	. = ..()
 	component_parts = list()
-	component_parts += new /obj/item/weapon/circuitboard/destructive_analyzer(src)
-	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
-	component_parts += new /obj/item/weapon/stock_parts/manipulator(src)
-	component_parts += new /obj/item/weapon/stock_parts/micro_laser(src)
+	component_parts += new /obj/item/weapon/circuitboard/destructive_analyzer(null)
```

Comment:
why create components in nullspace?

Reply:
In this case my bad but in case of microwave that was an intended feature. Microwave doesn't have any specific storage and remove stuff from contents